protocol/messenger.go

```diff
@@ -3059,12 +3059,26 @@ func (m *Messenger) handleRetrievedMessages(chatWithMessages map[transport.Filte
 
 	logger := m.logger.With(zap.String("site", "RetrieveAll"))
 
+	adminCommunitiesChatIDs, e := m.communitiesManager.GetAdminCommunitiesChatIDs()
+	if e != nil {
+		logger.Info("failed to retrieve admin communities", zap.Error(e))
+	}
+
 	for filter, messages := range chatWithMessages {
 		var processedMessages []string
 		for _, shhMessage := range messages {
 			logger := logger.With(zap.String("hash", types.EncodeHex(shhMessage.Hash)))
 			// Indicates tha all messages in the batch have been processed correctly
 			allMessagesProcessed := true
+
+			if adminCommunitiesChatIDs[filter.ChatID] {
```

Comment:
Would it maybe be safer to have this driven by some config, at least for now?

Reply:
We can put it behind the `HistoryArchiveSupportEnabled` flag as well. I just thought that, if we were to roll this out slowly, it' d be good if community owners would start accruing waku messages that can then be turned into archives as soon as they turn it on.

Happy to change this though. Just means that, if the protocol is turned off, no waku messages for history archives will be stored, so they'll be missed in the archives when they'll be created.